frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -300,48 +364,178 @@ function ReaderBookmark:onShowBookmark()
+            local temp = bookmark.bookmarks[i]
+            -- we need this reference to the unreversed, original bookmarks for deleting or renaming them:
+            temp.index = i
+            table.insert(bookmarks, temp)
+        end
+        local current_bookmark = 1
+        local page = ""
+        for nr, bmk in ipairs(bookmarks) do
+            if item.index == bmk.index then
+                current_bookmark = nr
+                -- grep the "page" string and page number:
+                page = item.text:match("^([A-Za-z]+ [0-9]+)")
+                if page == nil then
+                    page = ""
+                end
+                break
+            end
+        end
+        local title = T(_("Bookmark details  -  %1/%2"), tostring(current_bookmark), tostring(#bookmarks))
+        if page ~= "" then
+            title = title .. "  -  " .. string.lower(page)
+        end
+        local first_button_row = {
+            {
+                text = _("Go to"),
+                callback = function()
+                    UIManager:close(self.textviewer)
+                    UIManager:close(bookmark.bookmark_menu)
+                    bookmark.ui.link:addCurrentLocationToStack()
+                    bookmark:gotoBookmark(item.page)
+                end,
+            }
+        }
+        if bookmark.search_value ~= "" then
+            table.insert(first_button_row, {
+                text = "<",
+                callback = function()
+                    bookmark:search(bookmark.search_value, bookmarks, current_bookmark, bm_menu, self.textviewer, "previous")
+                end,
+            })
+        end
+        table.insert(first_button_row, {
+            text = _("Search"),
+            callback = function()
+                local arguments = {
+                    title = _("Search bookmark"),
+                    value = bookmark.search_value,
+                    hint = _("Needle"),
+                    callback = function(query)
+                        bookmark:search(string.lower(query), bookmarks, current_bookmark, bm_menu, self.textviewer, "forward")
+                    end,
+                    save_button_text = _("Search")
+                }
+                self:prompt(arguments)
+            end,
+        })
+        if bookmark.search_value ~= "" then
+            table.insert(first_button_row, {
+                text = ">",
+                callback = function()
+                    bookmark:search(bookmark.search_value, bookmarks, current_bookmark, bm_menu, self.textviewer, "forward")
+                end,
+            })
+        end
+        table.insert(first_button_row, {
+            text = _("Rename"),
+            callback = function()
+                local org_item = bookmark.bookmarks[item.index]
+                bookmark:renameBookmark(org_item)
+                UIManager:close(self.textviewer)
+            end,
+        })
+        table.insert(first_button_row, {
+            text = _("Remove"),
```

Comment:
For the order in the first row, I'd rather see the most destructive/rarely used on the left:
`Remove | Rename | < | Search | > | Go to`